tracer/test/Datadog.Trace.Tests/Util/QueryStringHelperTests.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="QueryStringHelperTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.Web;
+using Datadog.Trace.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util;
+
+public class QueryStringHelperTests
+{
+    [Fact]
+    public void GivenADangerousQueryString_WhenGetQueryString_HelperAvoidsException()
+    {
+        var request = new HttpRequest("file", "http://localhost/benchmarks", "data=<script>alert(1)</script>");
```

Comment:
Awesome, thanks for repro-ing this! It might be worth trying to create the same thing for .NET Core, just to _prove_ it's fine 🙂